### PR TITLE
Allow guest users to see the comments in CommentsContainer

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

**Review by bdach — DISMISSED** [review]
This can't be even directly tested right now. The fetch itself could be (by opening the game, logging out, and opening the test browser), but the refetch on logout cannot.

File: osu.Game/Overlays/Comments/CommentsContainer.cs [hunk 1]
```diff
@@ -113,6 +113,7 @@ private void load(OverlayColourProvider colourProvider)
 
         protected override void LoadComplete()
         {
+            api.LocalUser.BindValueChanged(_ => refetchComments());
```
Comment (iiSaLMaN) [inline osu.Game/Overlays/Comments/CommentsContainer.cs]: if I'm not missing something here, that's a memory leak?

**Review by frenzibyte — CHANGES_REQUESTED** [review]

**Comment by EVAST9919** [comment]
@bdach I can just a test step with `LocalUser.TriggerChange()`, but not sure about that

File: osu.Game.Tests/Visual/Online/TestSceneCommentsContainer.cs [hunk 2]
```diff
@@ -37,23 +39,29 @@ public class TestSceneCommentsContainer : OsuTestScene
...
             {
                 scroll.Clear();
-                scroll.Add(comments = new CommentsContainer());
+                scroll.Add(comments = new TestCommentsContainer());
             });
         }
+
+        private class TestCommentsContainer : CommentsContainer
+        {
+            public new Bindable<User> User => base.User;
```
Comment (iiSaLMaN) [inline osu.Game.Tests/Visual/Online/TestSceneCommentsContainer.cs]: no need to expose a bindable bound to a source you can easily retrieve.

**Review by frenzibyte — CHANGES_REQUESTED** [review]

File: osu.Game.Tests/Visual/Online/TestSceneCommentsContainer.cs (hunk 2 above)
Comment (smoogipoo) [reply osu.Game.Tests/Visual/Online/TestSceneCommentsContainer.cs]: i'm ok with this for now

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**